Userland/Applications/TextEditor/main.cpp

```diff
@@ -88,13 +88,61 @@ int main(int argc, char** argv)
     text_widget.initialize_menubar(menubar);
     app->set_menubar(menubar);
 
-    if (file_to_edit)
-        text_widget.open_file(file_to_edit);
-    else
+    // If a colon exists in the file name argument, the user may have specified line and column numbers.
+    // First check if the file exists before doing any further processing. If so, open it.
+    // Otherwise, split apart the line and column numbers from the file name, open the file, and jump to the specified line/column.
+
+    if (file_to_edit) {
+        if (StringView(file_to_edit).contains(':')) {
+            if (Core::File::exists(file_to_edit)) {
+                // A file exists with the full specified name, open it without doing any further processing.
+                text_widget.open_file(file_to_edit);
+            } else {
```

Comment:
This is a lot of nesting. Let's find a way to write this code so we don't have to nest as much :)

How about we move this logic to a function that takes the `file_to_edit` string and turns it into a `{ file_name, Optional<size_t> line, Optional<size_t> column }`?

Reply:
:thumbsup:
I've turned this into a `FileArgument` class which takes the `file_to_edit` string as does as you described, does this look ok?